docs/upgrading-to-v0.10.md

```diff
@@ -0,0 +1,75 @@
+# Upgrading to Ark v0.10
+
+## Overview
+
+Ark v0.10 includes a number of breaking changes. Below, we outline what those changes are, and what steps you should take to ensure
+a successful upgrade from prior versions of Ark.
+
+## Breaking Changes
+
+### Switch from Config CRD to BackupStorageLocation, VolumeSnapshotLocation, and server flags
+
+Prior to v0.10, Ark used a `Config` CRD to capture information about your backup storage and persistent volume providers, as well
+some miscellaneous Ark settings. In v0.10, we've eliminated this CRD and replaced it with:
+
+- A [BackupStorageLocation][1] CRD to capture information about where to store your backups
+- A [VolumeSnapshotLocation][2] CRD to capture information about where to store your persistent volume snapshots
+- Command-line flags for the `ark server` command (run by your Ark deployment) to capture miscellaneous Ark settings
+
+When upgrading to v0.10, you'll need to transfer the configuration information that you currently have in the `Config` CRD
+into the above. We'll cover exactly how to do this below.
+
+### Reorganization of data in object storage
+
+We've made [changes to the layout of data stored in object storage][3] for simplicity and extensibility. You'll need to
+rearrange any pre-v0.10 data as part of the upgrade. We've provided a script to help with this.
+
+## Step-by-Step Upgrade Instructions
+
+1. Scale down your existing Ark deployment:
+```bash
+kubectl scale -n heptio-ark deploy/ark --replicas 0
+```
+
+2. Re-apply the `00-prereqs.yaml` file to create new CRDs:
+```bash
+kubectl apply -f examples/common/00-prereqs.yaml
+```
+
+3. Create a [BackupStorageLocation][1] resource based on the examples provided in the `examples/` directory for your platform, using information from
+the existing `Config` resource as necessary.
+
+4. Create a [VolumeSnapshotLocation][2] resource based on the examples provided in the `examples/` directory for your platform, using information from
+the existing `Config` resource as necessary.
+
+5. Perform the one-time object storage migration detailed [here][3].
+
+6. In your Ark deployment YAML (see the `examples/` directory for samples), specify any non-default settings as args to the `ark server` command:
+
+| Flag | Default Value |
+| ---- | ------------- |
+| `--backup-sync-period` | 1m |
+| `--restic-timeout` | 1h |
+| `--restore-only` | false |
+
+7. In your Ark deployment YAML, define your default `BackupStorageLocation` and `VolumeSnapshotLocations`  as args to the `ark server` command:
```

Comment:
```suggestion
7. Still in your Ark deployment YAML, define the following as args to the `ark server` command:
- change `BackupStorageLocation` from the preset `default` name to another name if you would like to; this is the name of the location your backup will automatically be stored at so you don't need to specify it when you create a backup. If you want to use this default feature, it needs to match the name of at least 1 `BackupStorageLocation` CRD.
- if you will be backing up PVs, add `VolumeSnapshotLocations` (1 per provider, ex: `provider1:location-01,provider2:location-02,...`)
```

Reply:
```suggestion
7. In your Ark deployment YAML, define your default `BackupStorageLocation` and `VolumeSnapshotLocations` as args to the `ark server` command:
```

Reply:
Ah one thing I almost forgot to mention: in the deployment yaml under each provider's example folder, I think we should add at least an example for the BSL (w/ a name that is not `default` since that name is already the preset name in ark server), maybe even a couple for VSL. What do you think?

Reply:
Torn on this - the examples show the structure of the CRD. For most people, they're going to apply the example YAML file for their provider and only have 1 backupstoragelocation, much like how things were prior to v0.10. If we provide other examples, I think it will lead to confusion as to what file(s) to apply.

Reply:
I `s/define/may define` to maybe reduce the potential for misunderstanding that you `must` define those args.